restcomm.android.client.sdk/src/main/java/org/mobicents/restcomm/android/client/sdk/RCConnection.java

```diff
@@ -560,10 +563,19 @@ void initializeWebrtc(boolean videoEnabled)
         for (String permission : MANDATORY_PERMISSIONS) {
             if (context.checkCallingOrSelfPermission(permission) != PackageManager.PERMISSION_GRANTED) {
                 logAndToast("Permission " + permission + " is not granted");
-                // TODO: return error to RCConnection listener
+                listener.onPermissionNotGranted(this, permission);
```

Comment:
@mehturt: Please prefer using existing error reporting, by utilizing onDisconnected(connection, errorCode, errorText), as can be seen in line 510 above, instead of introducing a new callback (i.e. onPermissionGranted). Reason for this is that we want to be compatible with [Twilio's Android Client API](https://www.twilio.com/docs/api/client/android/ConnectionListener), plus we keep the API simpler and expect RCConnection errors from a single interface.